node/src/blockchain/blockchain_bridge.rs

```diff
@@ -300,84 +300,116 @@ impl BlockchainBridge {
         &mut self,
         msg: RetrieveTransactions,
     ) -> Box<dyn Future<Item = (), Error = String>> {
-        let start_block_nbr = match self.persistent_config.start_block() {
-                Ok (sb) => sb,
-                Err (e) => panic! ("Cannot retrieve start block from database; payments to you may not be processed: {:?}", e)
+        let (start_block, max_block_count) = {
+            let persistent_config_lock = self
+                .persistent_config_arc
+                .lock()
+                .expect("Unable to lock persistent config in BlockchainBridge");
+            let start_block_nbr = match persistent_config_lock.start_block() {
+                Ok(Some(sb)) => sb,
+                Ok(None) => u64::MAX,
+                Err(e) => panic!("Cannot retrieve start block from database; payments to you may not be processed: {:?}", e)
             };
-        let max_block_count = match self.persistent_config.max_block_count() {
-            Ok(Some(mbc)) => mbc,
-            _ => u64::MAX,
+            let max_block_count = match persistent_config_lock.max_block_count() {
+                Ok(Some(mbc)) => mbc,
+                _ => u64::MAX,
+            };
+            (start_block_nbr, max_block_count)
         };
-
-        let fallback_start_block_number =
-            calculate_fallback_start_block_number(start_block_nbr, max_block_count);
-        let start_block = BlockNumber::Number(start_block_nbr.into());
-        let received_payments_subs_ok_case = self
+        let logger = self.logger.clone();
+        let fallback_next_start_block_number =
+            Self::calculate_fallback_start_block_number(start_block, max_block_count);
+        let received_payments_subs = self
             .received_payments_subs_opt
             .as_ref()
             .expect("Accountant is unbound")
             .clone();
-        let received_payments_subs_error_case = received_payments_subs_ok_case.clone();
+        let persistent_config_arc = self.persistent_config_arc.clone();
 
         Box::new(
             self.blockchain_interface
                 .retrieve_transactions(
                     start_block,
-                    fallback_start_block_number,
+                    fallback_next_start_block_number,
                     msg.recipient.address(),
                 )
                 .map_err(move |e| {
-                    let received_payments_error =
-                        match BlockchainBridge::extract_max_block_count(e.clone()) {
-                            Some(max_block_count) => {
-                                ReceivedPaymentsError::ExceededBlockScanLimit(max_block_count)
+                    if let Some(max_block_count) =
+                        BlockchainBridge::extract_max_block_count(e.clone())
+                    {
+                        match persistent_config_arc
+                            .lock()
+                            .expect("Unable to lock persistent config in BlockchainBridge")
```

Comment:
There could be a more precise message like: "Mutex with persistent config in BlockchainBridge was poisoned".